src/odemis/driver/technolution.py

```diff
@@ -834,130 +852,168 @@ def __init__(self, name, role, parent, **kwargs):
         # Use almost_equal to handle floating point errors.
         if not almost_equal(remainder_scanning_time, 0, rtol=0, atol=1e-10):
-            # Allow the descanner to finish scanning the last pixels by adding one setpoint with the scan gain value.
-            scanning_points = numpy.hstack((scanning_points, scan_gain_bits))
-            # Note: For now the extra setpoint to compensate for the remaining scanning time has a value equal to the
-            # scan gain. Because of the inertia of the mirror it is expected that this is the fastest solution
-            # which will not significantly disturb the trajectory of the mirror.
+            # add one setpoint with value same as end of scanning ramp
+            scanning_points = numpy.hstack((scanning_points, scan_end))  # [bits]
 
         # Calculation of the flyback points:
-        # Round up so that if the physical flyback time (physical restricted time for the mirror to move back to its
-        # original position) is not a whole multiple of the descan period the flyback points allow enough time for the
-        # descanner to move back.
-        number_flyback_points = math.ceil(self.physicalFlybackTime / descan_period)
-        flyback_points = scan_offset_bits + numpy.zeros(number_flyback_points)
+        # Check if the physical flyback time (time the mirror needs to move back to the start position
+        # of the line scan) is a whole multiple of the descan period. If not, round up.
+        number_flyback_points = math.ceil(self.physicalFlybackTime / descan_period)  # [sec/sec = no unit]
+        # convert flyback setpoints into bits; should be at the same level as the start of the ramp (=offset)
+        flyback_points = scan_start + numpy.zeros(number_flyback_points)  # [bits]
 
-        setpoints = numpy.concatenate((scanning_points, flyback_points))
+        setpoints = numpy.concatenate((scanning_points, flyback_points))  # [bits]
 
-        # Converting to an integer using numpy.floor() is more correct than int() because the latter one rounds down
-        # towards 0, so negative values are not treated the same as positive values, which makes values not uniformly
-        # distributed.
-        return numpy.floor(setpoints).astype(int).tolist()
+        # Setpoints need to be integers when send to the ASM. First round down found setpoints to next integer
+        # then convert to int type.
+        # For consistency in the rounding use numpy.floor() to always round down (also for negative numbers):
+        # e.g. int(-3.4) = -3 vs. numpy.floor(-3.4) = -4.
+        setpoints = numpy.floor(setpoints).astype(int).tolist()  # [bits]
+
+        # mapping from a.u. to bits is symmetrically around 0, whereas the range in bits that is accepted by the ASM is
+        # not symetrically around 0 ([-32768, 32767]). Clip 2**15 = 32768 by one bit to 32767 bit.
+        setpoints = [p if p != BIT_RANGE[1] else (BIT_RANGE[1] - 1) for p in setpoints]
```

Comment:
As you have a numpy array just before, I'd suggest to make it faster by using a numpy operation:
```python
setpoints = numpy.floor(setpoints).astype(int)
setpoints = numpy.minimum(setpoints, BIT_RANGE[1] - 1)
return setpoints.lolist()
```